tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -1529,6 +1606,54 @@ pub(super) mod test {
         run_test(subscriber, make_writer, expected);
     }
 
+    #[test]
+    fn with_line_number_and_file_name() {
+        let make_writer = MockMakeWriter::default();
+        let subscriber = crate::fmt::Collector::builder()
+            .with_writer(make_writer.clone())
+            .with_filename(true)
+            .with_line_number(true)
+            .with_level(false)
+            .with_ansi(false)
+            .with_timer(MockTime);
+
+        let expected = Regex::new("^fake time tracing_subscriber::fmt::format::test: tracing-subscriber/src/fmt/format/mod.rs: [0-9]+: hello\n$").unwrap();
```

Comment:
it looks like these tests fail on Windows, because Windows formats paths using `\` rather than `/`. see the CI failure: https://github.com/tokio-rs/tracing/runs/4477194846?check_suite_focus=true

perhaps what we should do instead is build the regex using the _actual_ path to this file, rather than hard-coding it. that way, these tests would pass regardless of the operating system's path separator character. also, we wouldn't have to update it if we moved the test into a different file. we could write something like this:
```suggestion
        let expected = Regex::new(concat!(
            "^fake time ",
             module_path!(),
             " :",
             file!(),
             ": [0-9]+: hello\n$"
         )).unwrap();
```

Reply:
actually, hmm, this won't quite work, since we'd also need to escape any backslash characters in the path on windows, but we could do that by using `format!` rather than `concat!`, and using `str::replace` to replace any `\`s in the file path with `\\`